File: backend/demo/src/main/java/com/example/models/Post.java
package com.example.models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;

import java.time.LocalDateTime;
import java.util.ArrayList;

@Entity
public class Post {

    @Id
    @GeneratedValue
    private Long id;

    private Long posterId;

    private String title;
    private String message;
    private LocalDateTime createdAt;
    private Long upvotes;
    private Long downvotes;

    private ArrayList<String> comments;
}

File: frontend/src/Posts/Post.ts


export interface post {
    id: number;
    authorId: string;
    title: string;
    message: string;
    upVotes: number;
    category: string;
    comments: string[];
    createdAt: Date;
    userHasUpvoted: boolean;

}
export interface createPostRequest {
    boardId?: number;
    title: string;
    message: string;
    category: string;
    
}